### DISCUSSION TITLE
avcpp requiers SSL. How to properly link?

### DISCUSSION THREAD

**Discussion by dsazonoff** [microsoft/vcpkg#35568] [Q&A] [answered]
Hello, I'm trying to use avcpp in my project, but there is an issue. Project linkage fails with a lot of errors:
```
ld: Undefined symbols:
  _SSLClose, referenced from:
      _tls_close in libavformat.a[x86_64][457](tls_securetransport.o)
  _SSLCopyPeerTrust, referenced from:
      _tls_open in libavformat.a[x86_64][457](tls_securetransport.o)
...
``` 

Any ideas, how to properly link with avcpp? Or it's a port issue?
My environment: intel-based mac (sonoma)

main.cpp:
```c++
#include <av.h>
#include <format.h>

int main()
{
    av::init();
    auto in = av::InputFormat{"/Users/saz/temp.mp3"}; // <--- Remove for successful build
    return 0;
}
``` 

CMakeLists.txt
```cmake
cmake_minimum_required(VERSION 3.27)
project(avcpp_01)

set(CMAKE_CXX_STANDARD 20)

find_package(Threads REQUIRED) # <--- without it cmake fails
find_package(avcpp CONFIG REQUIRED)

add_executable(avcpp_01 main.cpp)
target_link_libraries(avcpp_01 PRIVATE avcpp::avcpp-static avcpp::FFmpeg)
```

vcpkg.json
```json
{
  "name": "avcpp-01",
  "version-string": "2023.0.1",
  "dependencies": [
    {
      "name": "avcpp"
    }
  ]
}
```

**Comment by dsazonoff** [comment]
Seems that `target_link_libraries(avcpp_01 PRIVATE "-framework Security")` is missing in avcpp-config.cmake

**Reply by dg0yt** [reply, marked as answer]
AFAICS avcpp comes with a cmake find module for ffmpeg. That modules uses a common pattern: using pkg-config to locate the ffmpeg root, but not using pkg-config to discover transitive usage requirements such as link libraries. And so it breaks for static linkage.
